CMakeLists.txt

```diff
@@ -402,7 +402,7 @@ if(SINGULARITY_BUILD_CLOSURE)
     add_library(singularity-eos_Library)
     set_target_properties(singularity-eos_Library PROPERTIES OUTPUT_NAME singularity-eos)
     add_library(singularity-eos::singularity-eos_Library ALIAS singularity-eos_Library)
-    target_link_libraries(singularity-eos_Library INTERFACE singularity-eos_Common)
+    target_link_libraries(singularity-eos_Library PUBLIC singularity-eos_Common)
```

Comment:
I'm very confused by why this works but... :shrug:

Reply:
Me too, but I seem to recall linking issues with the wrong scope before. I _think_ this just gets the libraries later in the link command, but I'm not 100% sure. I defer to @rbberger

Reply:
You can think of this as telling CMake to add an explicit dependency to Kokkos (via Common) when compiling the library itself and to anything that links to it. It's a missing edge in the dependency graph that lets CMake determine the proper link line.